File: app/src/main/java/com/kksionek/gdzietentramwaj/DataSource/TramDataWrapper.java
package com.kksionek.gdzietentramwaj.DataSource;

import java.util.Map;

import io.reactivex.annotations.Nullable;

public class TramDataWrapper {
    public Map<String, TramData> tramDataHashMap;
    public Throwable throwable;
    public boolean loading;

    public TramDataWrapper(
            @Nullable Map<String, TramData> tramDataHashMap,
            @Nullable Throwable throwable,
            boolean loading) {
        this.tramDataHashMap = tramDataHashMap;
        this.throwable = throwable;
        this.loading = loading;
    }
}

File: app/src/main/java/com/kksionek/gdzietentramwaj/dataSource/TramDataWrapper.kt
package com.kksionek.gdzietentramwaj.dataSource

sealed class TramDataWrapper {
    data class Success(val tramDataHashMap: Map<String, TramData>) : TramDataWrapper()

    data class Error(val throwable: Throwable) : TramDataWrapper()

    object InProgress : TramDataWrapper()
}
